CMP1903M Object Oriented Programming A01 2223/CMP1903M A01 2223/CMP1903M A01 2223/Pack.cs

```diff
@@ -8,26 +9,89 @@ namespace CMP1903M_A01_2223
 {
     class Pack
     {
-        List<Card> pack;
+        
+        //Get, set values
+        public int typeOfShuffle { get; set; }
+        public int amount { get; set; }
+
+        //Initialise the card pack here
+        public static List<Card> _pack = new List<Card>();
+        public Card[] cardPack = _pack.ToArray();
 
         public Pack()
         {
-            //Initialise the card pack here
         }
-
-        public static bool shuffleCardPack(int typeOfShuffle)
+        public static bool shuffleCardPack(int typeOfShuffle, int[] cardPack)
         {
             //Shuffles the pack based on the type of shuffle
-
+            if(typeOfShuffle == 1)
+            {
+                Console.WriteLine("Fisher–Yates shuffle");
+                Random _random = new Random(); 
+                int n = cardPack.Length; 
+                //Will change all positions of cards with a random card
+                for(int i = 0; i < n; i++)
+                {
+                    int r = i + _random.Next(n - 1);
+                    int temp = cardPack[r];

```

Comment:
instead of int temp it should be Card temp. Because you should be shuffling an array or list of Card objects